magicgui/_tests/test_ qt.py

```diff
@@ -90,3 +101,75 @@ def test_set_categorical(qtbot):
     assert [w.itemText(i) for i in range(w.count())] == ["a", "b"]
     _qt.set_categorical_choices(w, (("a", 1), ("c", 3)))
     assert [w.itemText(i) for i in range(w.count())] == ["a", "c"]
+
+
+def test_magicfiledialog(qtbot):
+    """Test the MagicFileDialog class."""
+    filewidget = _qt.MagicFileDialog()
+
+    # check default values
+    assert filewidget.get_path() == Path('.')
+    assert filewidget.mode == _qt.FileDialogMode.OPTIONAL_FILE
+
+    # set the mode
+    filewidget.mode = _qt.FileDialogMode.EXISTING_FILES  # Enum input
+    assert filewidget.mode == _qt.FileDialogMode.EXISTING_FILES
+    filewidget.mode = "oPtioNal_FiLe"  # improper capitalization
+    assert filewidget.mode == _qt.FileDialogMode.OPTIONAL_FILE
+    filewidget.mode = "EXISTING_DIRECTORY"  # string input
+    assert filewidget.mode == _qt.FileDialogMode.EXISTING_DIRECTORY
+    with pytest.raises(ValueError):
+        filewidget.mode = 123  # invalid mode
```

Comment:
```suggestion
    with pytest.raises(ValueError):
        filewidget.mode = 123  # invalid mode
    with pytest.raises(ValueError):
        filewidget.mode = 'invalid_string'
```
(need to check for an invalid string... not just an invalid type)